### ISSUE TITLE
Readme: create dataset

### ISSUE THREAD

**Issue by tschoy** [tczuo/HF-FCN-for-Robust-Building-Extraction#3]
https://github.com/tczuo/HF-FCN-for-Robust-Building-Extraction#create-dataset

probably some typo?
python scripts/create_dataset_256.py -d data
python scripts/verify_dataset.py -d data/mass_buildings/lmdb/train_sat_256 

Also scripts/create_dataset_256.py last lines 120-127 have syntax error:
args.dataset'/mass_buildings/lmdb/train_map' probably missing a + sign.

**Comment by tczuo** [comment]
I'm sorry for this edit mistake, the correct command is following. I have fixed it in readme.txt.
python scripts/create_dataset_256.py  -d /data

**Comment by VisionU** [comment]
I'm sorry for this edit mistake, the correct command is following. I have
fixed it in readme.md.
python scripts/create_dataset_256.py -d /data

2017-05-02 18:16 GMT+08:00 tschoy <notifications@github.com>:

**Comment by PraveerSINGH** [comment]
Hey @tczuo  you need to correct the readme.md to -> 

`python scripts/create_dataset_256.py -d ./data
python scripts/verify_dataset.py -d ./data/mass_buildings/lmdb/train_sat`

And inside scripts/create_dataset_256.py, between lines 120-127, need to add + signs like ->

`args.dataset + '/mass_buildings/lmdb/train_map`

**Comment by VisionU** [comment]
@PraveerSINGH,  It's so strange that "+" signs are missed. Thanks for your  careful check and your attention for our paper. I have fixed this problem of my codes.

**Comment by tschoy** [comment]
Dear Tony,
Do you know what's the minimum GPU Memory does your code requires to predict or train? I have used a small 4GB GPU but it ran out of RAM even running the prediction. I'm trying to locate larger GPU cards, but it would be good to know the RAM requirement, is 12GB a must?
I'm also thinking about reducing the network and image size, but then I would lost the benefit of re-using your pre-trained weights.
Best Regards,Alex 

    On Thursday, May 18, 2017 11:29 AM, Tony <notifications@github.com> wrote:
 

 Closed #3.—

**Comment by VisionU** [comment]
@tschoy  If you would like to use 4GB RAM, please reduce the batch size from 18 to 6. This operation might loss some accuracy but you still can use my pre-trained weights.

**Comment by VisionU** [comment]
@tschoy  I suggest that you'd better not reduce the network and image size, just reduce batchsize.

**Comment by tschoy** [comment]
Dear Tony,
I have got a K80 GPU so I could run your program now. I have successfully run the prediction using 
python ../scripts/run_prediction.py  --model ../models/HF-FCN_Models/BasicNet/predict.prototxt --weight ../weights/HF-FCN_iter_12000.caffemodel --img_dir ..//data/mass_buildings/test/sat/

and the result seems to be identical with your paper. I have two problem now:
1. My goal is to apply this program on other aerial data. When I tried to test it on a different image, like the attached (out17.tiff), which is for Massachusetts from Google Map (Level 17), and is the same region as in Mnih's test data, and about the same scale. However, my result is not as good as testing on Mnih data (23729035_15.tiff), my result is attached as attached image 

(pred_0_out17.tiff.png). Do you mind total a look and may be offer some clue as in what I should do?
2. I also ran the solver using: 
nohup python solve.py&

And after running, I use the file BuildingDetection_iter_12000.caffemodel or BuildingDetection_iter_80000.caffemodel as weights, however I only get a completely gray image as pred_0_*.tiff.png.
Any help would be very appreciated! Thank you!
Alex Choy
 

    On Thursday, May 18, 2017 7:55 PM, Tony <notifications@github.com> wrote:
 

 @tschoy I suggest that you'd better not reduce the network and image size, just reduce batchsize.—

**Comment by tschoy** [comment]
Hi Tony, 
The attached in this email is my run using weight from your GitHub on the Mind data, it looks very nice. Just when I use google map image I got much less exciting result and not sure what to do. Thanks a lot for you help.



 

    On Monday, June 26, 2017 5:59 PM, Alex <tschoy@yahoo.com> wrote:
 

 Dear Tony,
I have got a K80 GPU so I could run your program now. I have successfully run the prediction using 
python ../scripts/run_prediction.py  --model ../models/HF-FCN_Models/BasicNet/predict.prototxt --weight ../weights/HF-FCN_iter_12000.caffemodel --img_dir ..//data/mass_buildings/test/sat/

and the result seems to be identical with your paper. I have two problem now:
1. My goal is to apply this program on other aerial data. When I tried to test it on a different image, like the attached (out17.tiff), which is for Massachusetts from Google Map (Level 17), and is the same region as in Mnih's test data, and about the same scale. However, my result is not as good as testing on Mnih data (23729035_15.tiff), my result is attached as attached image 

(pred_0_out17.tiff.png). Do you mind total a look and may be offer some clue as in what I should do?
2. I also ran the solver using: 
nohup python solve.py&

And after running, I use the file BuildingDetection_iter_12000.caffemodel or BuildingDetection_iter_80000.caffemodel as weights, however I only get a completely gray image as pred_0_*.tiff.png.
Any help would be very appreciated! Thank you!
Alex Choy
 

    On Thursday, May 18, 2017 7:55 PM, Tony <notifications@github.com> wrote:
 

 @tschoy I suggest that you'd better not reduce the network and image size, just reduce batchsize.—

**Comment by VisionU** [comment]
Dear tschoy:
     Sorry for replying you so late.   Though I check your email carefully,
I couldn't find your attachment (out17.tiff, pred_0_out17.tiff.png). I
wonder that you forget uploading your results into this email. Could you
please check it and send the attachment directly?
[image: 内嵌图片 1]

Tony Zuo

2017-06-26 17:59 GMT+08:00 tschoy <notifications@github.com>:

> Dear Tony,
> I have got a K80 GPU so I could run your program now. I have successfully
> run the prediction using
> python ../scripts/run_prediction.py  --model ../models/HF-FCN_Models/BasicNet/predict.prototxt
> --weight ../weights/HF-FCN_iter_12000.caffemodel --img_dir
> ..//data/mass_buildings/test/sat/
>
> and the result seems to be identical with your paper. I have two problem
> now:
> 1. My goal is to apply this program on other aerial data. When I tried to
> test it on a different image, like the attached (out17.tiff), which is for
> Massachusetts from Google Map (Level 17), and is the same region as in
> Mnih's test data, and about the same scale. However, my result is not as
> good as testing on Mnih data (23729035_15.tiff), my result is attached as
> attached image
>
> (pred_0_out17.tiff.png). Do you mind total a look and may be offer some
> clue as in what I should do?
> 2. I also ran the solver using:
> nohup python solve.py&
>
> And after running, I use the file BuildingDetection_iter_12000.caffemodel
> or BuildingDetection_iter_80000.caffemodel as weights, however I only get
> a completely gray image as pred_0_*.tiff.png.
> Any help would be very appreciated! Thank you!
> Alex Choy
>
>

**Comment by ZHAOZHIHAO** [comment]
Hi, tczuo

You wrote "python scripts/create_dataset_256.py  -d /data" in Readme, but I think it should be "python scripts/create_dataset_256.py  -d data" without the symbol '/' as tschoy said. May you modify the readme for other developers for reproduction? Thanks.

Zhihao Zhao
